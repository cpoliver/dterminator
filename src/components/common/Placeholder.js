import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, Text, View } from 'react-native';
import { Icon } from 'react-native-elements';

const Placeholder = ({ iconName }) => (
  <View style={styles.container}>
    <Icon color="#25c7ee" size={92} type="material-community" name={iconName} />
    <Text style={styles.text}>coming soon</Text>
  </View>
);

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center'
  },
  text: {
    color: '#25c7ee',
    fontSize: 20
  }
});

Placeholder.propTypes = {
  iconName: PropTypes.string.isRequired
};

export { Placeholder };
