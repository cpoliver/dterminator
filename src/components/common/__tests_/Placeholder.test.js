import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Placeholder } from '../Placeholder';

const render = () => shallow(
  <Placeholder iconName="account" />
);

describe('the Placeholder component', () => {
  it('should match the snapshot', () => {
    const component = render();

    expect(toJson(component)).toMatchSnapshot();
  });
});
