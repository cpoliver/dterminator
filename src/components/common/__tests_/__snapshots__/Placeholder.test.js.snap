// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Placeholder component should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Icon
    color="#25c7ee"
    name="account"
    raised={false}
    reverse={false}
    reverseColor="white"
    size={92}
    type="material-community"
    underlayColor="white"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#25c7ee",
        "fontSize": 20,
      }
    }
  >
    coming soon
  </Text>
</View>
`;
